src/operator.js

```diff
@@ -0,0 +1,72 @@
+/** ************************************************************************
+ *  (C) Copyright Mojaloop Foundation 2020                                *
+ *                                                                        *
+ *  This file is made available under the terms of the license agreement  *
+ *  specified in the corresponding source code repository.                *
+ *                                                                        *
+ *  ORIGINAL AUTHOR:                                                      *
+ *       Yevhen Kyriukha <yevhen.kyriukha@modusbox.com>                   *
+ ************************************************************************* */
+
+const k8s = require('@kubernetes/client-node');
+const { Logger } = require('@mojaloop/sdk-standard-components').Logger;
+const ejs = require('ejs');
+const config = require('./config');
+const { createReportHandler } = require('./handlers');
+
+const logger = new Logger();
+
+const kc = new k8s.KubeConfig();
+kc.loadFromDefault();
+
+// const k8sApiMC = kc.makeApiClient(k8s.CustomObjectsApi);
+
+// Listen for events or notifications and act accordingly
+const watch = new k8s.Watch(kc);
+
+async function onEvent(reportData, phase, apiObj) {
+    logger.info(`Received event in phase ${phase} for the resource ${apiObj?.metadata?.name}`);
+
+    const name = apiObj?.metadata?.name;
+    const path = apiObj.spec.endpoint.path.toLowerCase();
+    const { handlerMap, renderMap, pathMap } = reportData;
+    if (name) {
+        if (phase === 'ADDED') {
+            renderMap[name] = ejs.compile(apiObj.spec.template);
+            handlerMap[path] = createReportHandler(renderMap[name], apiObj.spec);
+            pathMap[path] = name;
+        } else if (phase === 'MODIFIED') {
+            renderMap[name] = ejs.compile(apiObj.spec.template);
+            handlerMap[path] = createReportHandler(renderMap[name], apiObj.spec);
+            pathMap[path] = name;
+        } else if (phase === 'DELETED') {
+            delete handlerMap[path];
+            delete renderMap[name];
+            delete pathMap[path];
+        } else {
+            logger.warn(`Unknown event type: ${phase}`);
+        }
+    }
+}
+
+function watchResource(reportData) {
+    return watch.watch(
+        // eslint-disable-next-line max-len
```

Comment:
It's possible to disable linting long strings. I'd be inclined to, personally.
https://eslint.org/docs/rules/max-len#options
Example: https://github.com/mojaloop/finance-portal-v2-ui/blob/20e16aaa1c9f9fbde67b85988ce1f42187f75a19/.eslintrc.js#L52